ci/kind/test-netpol-v2-conformance-kind.sh

```diff
@@ -0,0 +1,181 @@
+    --ip-family)
+    ipfamily="$2"
+    shift 2
+    ;;
+    --encap-mode)
+    mode="$2"
+    shift 2
+    ;;
+    --setup-only)
+    setup_only=true
+    shift
+    ;;
+    --cleanup-only)
+    cleanup_only=true
+    shift
+    ;;
+    --test-only)
+    test_only=true
+    shift
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+if [[ $cleanup_only == "true" ]];then
+  $TESTBED_CMD destroy kind
+  exit 0
+fi
+
+trap "quit" INT EXIT
+
+manifest_args="$manifest_args --verbose-log --coverage"
+if [ -n "$feature_gates" ]; then
+  manifest_args="$manifest_args --feature-gates $feature_gates"
+fi
+
+IMAGE_LIST=("registry.k8s.io/e2e-test-images/agnhost:2.43" \
+            "antrea/antrea-ubuntu-coverage:latest")
+
+for image in "${IMAGE_LIST[@]}"; do
+    for i in $(seq 3); do
+        docker pull $image && break
+        sleep 1
+    done
+done
+
+printf -v IMAGES "%s " "${IMAGE_LIST[@]}"
+
+function setup_cluster {
+  args=$1
+
+  if [[ "$ipfamily" == "v6" ]]; then
+    args="$args --ip-family ipv6 --pod-cidr fd00:10:244::/56"
+  elif [[ "$ipfamily" != "v4" ]]; then
+    echoerr "invalid value for --ip-family \"$ipfamily\", expected \"v4\" or \"v6\""
+    exit 1
+  fi
+
+  echo "creating test bed with args $args"
+  eval "timeout 600 $TESTBED_CMD create kind $args"
+}
+
+function run_test {
+  current_mode=$1
+
+  # Install the network-policy-api CRDs in the kind cluster
+  kubectl apply -f https://github.com/kubernetes-sigs/network-policy-api/releases/download/"$api_version"/install.yaml
+  echo "Generating Antrea manifest with args $manifest_args"
+  $YML_CMD --encap-mode $current_mode $manifest_args | kubectl apply -f -
+  $YML_CMD --encap-mode $current_mode $manifest_args | docker exec -i kind-control-plane dd of=/root/antrea.yml
+
+  kubectl rollout status --timeout=1m deployment.apps/antrea-controller -n kube-system
+  kubectl rollout status --timeout=1m daemonset/antrea-agent -n kube-system
+
+  sleep 1
+  export KUBERNETES_CONFORMANCE_TEST=y
+  export KUBE_CONTAINER_RUNTIME=remote
+  export KUBE_CONTAINER_RUNTIME_ENDPOINT=unix:///run/containerd/containerd.sock
+  export KUBE_CONTAINER_RUNTIME_NAME=containerd
+
+  # TODO: use https://github.com/kubernetes-sigs/network-policy-api when conformance test config timeout and go dependency is fixed
+  git clone https://github.com/Dyanngg/network-policy-api.git
+  pushd network-policy-api/conformance
+  go mod download
+  go test -v --debug=true -timeout=15m
+  popd
+}
+
+if [[ "$mode" == "" ]] || [[ "$mode" == "encap" ]]; then
+  echo "======== Test encap mode =========="
+  if [[ $test_only == "false" ]];then
+    setup_cluster "--images \"$IMAGES\""
+  fi
+  run_test encap
+fi
+if [[ "$mode" == "" ]] || [[ "$mode" == "noEncap" ]]; then
+  echo "======== Test noEncap mode =========="
+  if [[ $test_only == "false" ]];then
+    setup_cluster "--images \"$IMAGES\""
+  fi
+  run_test noEncap
+fi
```

Comment:
I feel like to simplify things in this PR, we should stick to encap mode. Typically, testing NP enforcement in different encap modes is not super useful.

Reply:
Done